File: LONG CHALLENGE/MAY222C/Subsequence-Numbers_Divisible_by_7.py
from sys import stdin, stdout
from collections import deque,Counter,OrderedDict,defaultdict,ChainMap,namedtuple
from itertools import permutations,combinations
import heapq
from bisect import bisect_left,bisect_right


def get_abcd(): return map(int, stdin.readline().strip().split())
def get_input(): return stdin.readline().strip()
def get_list(): return list(map(int, stdin.readline().strip().split()))
def get_str(): return list(map(str, stdin.readline().strip().split()))
def get_string(): return stdin.readline().strip()
def get_2str(a): return [stdin.readline().strip() for i in range(a)]
def get_2int(a): return [int(stdin.readline().strip().split()) for i in range(a)]
def show_output(val): stdout.write(str(val)+'\n')


mod=1000000007

try:


    def solve():
        ans=[[0 for i in range(7) ]for j in range(n+1)]
        ans[0][0]=1
        for i in range(n):
            s=len(str(l1[i]))
            # print(s)
            power=1
            while(s) :
                power*=10
                s-=1
            # print(power)
            for j in range(7):
                temp=(j*power+l1[i])%7
                ans[i+1][j]=(ans[i+1][j]+ans[i][j])%mod
                ans[i+1][temp]=(ans[i+1][temp]+ans[i][j])%mod
        show_output((ans[n][0]-1+mod)%mod)


    if __name__=="__main__":
        for _ in range(int(stdin.readline())):
            n=get_input()
            n=int(n)
            l1=get_list()
            solve()
except:
    pass

File: LONG CHALLENGE/MAY222C/Subsequence-Numbers_Divisible_by_7.cpp
#include<bits/stdc++.h>
#define ll long long
#define f(i,n) for(int i=0;i<n;i++)
#define r(i,n) for(int i=n-1;i>0;i--)
using namespace std;


void solve()
{
    ll n;
    cin>>n;
    ll mod=1e9+7;
    vector<int> arr(n,0);
    f(i,n) cin>>arr[i];
    
    vector<vector<int>> ans(n+1,vector<int>(7));
    ans[0][0]=1;
    f(i,n){
        ll l=(ll)to_string(arr[i]).size();
        ll power=1;
        while(l--) power*=10;
        f(j,7){
            ll temp=(j*power+arr[i])%7;
            ans[i+1][j]=(ans[i+1][j]+ans[i][j])%mod;
            ans[i+1][temp]=(ans[i+1][temp]+ans[i][j])%mod;
        }
    } 
    cout<<(ans[n][0]-1+ mod)%mod<<endl;
}

int main()
{
    ll testcases;
    cin>>testcases;
    while(testcases--)
    {
        solve();
    }
    return 0;
}